Modules/Session/classes/class.ilObjSession.php

```diff
@@ -846,6 +846,97 @@ public function handleAutoFill()
 
 	// cat-tms-patch start
 	/**
+	 * Checks whether this object is a child element of a course object.
+	 * If there is an group object first in tree it returns false.
+	 *
+	 * @return int | null
+	 */
+	public function isCourseOrCourseChild($ref_id)
+	{
+		global $DIC;
+		$g_tree = $DIC->repositoryTree();
+		$tree = array_reverse($g_tree->getPathFull($ref_id));
+		foreach ($tree as $leaf)
+		{
+			if($leaf['type'] === "crs")
+			{
+				return $leaf['ref_id'];
+			}
+		}
```

Comment:
Wenn es wichtig ist, dass die Session in einer Gruppe liegt, dann sollte hier auch irgendwie darauf geprüft werden.